src/misc/lv_grad.c

```diff
@@ -0,0 +1,108 @@
+/**
+ * @file lv_grad.c
+ *
+ */
+
+/*********************
+ *      INCLUDES
+ *********************/
+#include "lv_grad.h"
+
+/*********************
+ *      DEFINES
+ *********************/
+
+/**********************
+ *      TYPEDEFS
+ **********************/
+
+/**********************
+ *  STATIC PROTOTYPES
+ **********************/
+
+/**********************
+ *  STATIC VARIABLES
+ **********************/
+
+/**********************
+ *      MACROS
+ **********************/
+
+/**********************
+ *   GLOBAL FUNCTIONS
+ **********************/
+void lv_grad_init_stops(lv_grad_dsc_t * grad, const lv_color_t colors[], const lv_opa_t opa[],
+                        const uint8_t fracs[], int num_stops)
+{
+    LV_ASSERT(num_stops <= LV_GRADIENT_MAX_STOPS);
+    grad->stops_count = num_stops;
+    for(int i = 0; i < num_stops; i++) {
+        grad->stops[i].color = colors[i];
+        grad->stops[i].opa = opa != NULL ? opa[i] : LV_OPA_COVER;
+        grad->stops[i].frac = fracs != NULL ? fracs[i] : 255 * i / (num_stops - 1);
+    }
+}
+
+void lv_grad_horizontal_init(lv_grad_dsc_t * dsc)
+{
+    dsc->dir = LV_GRAD_DIR_HOR;
+}
+
+void lv_grad_vertical_init(lv_grad_dsc_t * dsc)
+{
+    dsc->dir = LV_GRAD_DIR_VER;
+}
+
+
+#if LV_USE_DRAW_SW_COMPLEX_GRADIENTS
+
+void lv_grad_linear_init(lv_grad_dsc_t * dsc, int32_t from_x, int32_t from_y, int32_t to_x, int32_t to_y,
```

Comment:
Should `from_x` and `from_y` be `lv_point_t from`? This way the `LV_USE_FLOAT` will also work.

Reply:
I'm about to move away from the point in the APIs because if the user x and y coordinates separately they need to create a point first. 

We could use `lv_value_precise_t` but  they values can be percentage too which is not supported by floats now.